python/cuspatial/cuspatial/tests/binpreds/binpred_test_dispatch.py

```diff
@@ -0,0 +1,535 @@
+        point_polygon,
+    ),
+    "polygon-polygon-same": (
+        """
+    x---x
+    |   |
+    |   |
+    |   |
+    x---x
+    """,
+        point_polygon,
+        point_polygon,
+    ),
+}
+
+point_point_dispatch_list = [
+    "point-point-disjoint",
+    "point-point-equal",
+]
+
+point_linestring_dispatch_list = [
+    "point-linestring-disjoint",
+    "point-linestring-point",
+    "point-linestring-edge",
+]
+
+point_polygon_dispatch_list = [
+    "point-polygon-disjoint",
+    "point-polygon-point",
+    "point-polygon-edge",
+    "point-polygon-in",
+]
+
+linestring_linestring_dispatch_list = [
+    "linestring-linestring-disjoint",
+    "linestring-linestring-same",
+    "linestring-linestring-touches",
+    "linestring-linestring-crosses",
+]
+
+linestring_polygon_dispatch_list = [
+    "linestring-polygon-disjoint",
+    "linestring-polygon-touch-point",
+    "linestring-polygon-touch-edge",
+    "linestring-polygon-overlap-edge",
+    "linestring-polygon-intersect-edge",
+    "linestring-polygon-intersect-inner-edge",
+    "linestring-polygon-point-interior",
+    "linestring-polygon-edge-interior",
+    "linestring-polygon-in",
+    "linestring-polygon-crosses",
+]
+
+polygon_polygon_dispatch_list = [
+    "polygon-polygon-disjoint",
+    "polygon-polygon-touch-point",
+    "polygon-polygon-touch-edge",
+    "polygon-polygon-overlap-edge",
+    "polygon-polygon-point-inside",
+    "polygon-polygon-point-outside",
+    "polygon-polygon-in-out-point",
+    "polygon-polygon-in-point-point",
+    "polygon-polygon-contained",
+    "polygon-polygon-same",
+]
+
+
+def object_dispatch(name_list):
+    """Generate a list of test cases for a given set of test names."""
+    for name in name_list:
+        yield (name, features[name][0], features[name][1], features[name][2])
+
+
+type_dispatch = {
+    # A dictionary of test cases for each geometry type combination.
+    # Still needs MultiPoint.
+    (Point, Point): object_dispatch(point_point_dispatch_list),
+    (Point, LineString): object_dispatch(point_linestring_dispatch_list),
+    (Point, Polygon): object_dispatch(point_polygon_dispatch_list),
+    (LineString, LineString): object_dispatch(
+        linestring_linestring_dispatch_list
+    ),
+    (LineString, Polygon): object_dispatch(linestring_polygon_dispatch_list),
+    (Polygon, Polygon): object_dispatch(polygon_polygon_dispatch_list),
+}
+
+
+def simple_test_dispatch():
+    """Generates a list of test cases for each geometry type combination."""
+    for types in type_dispatch:
+        generator = type_dispatch[types]
+        for test_name, test_description, lhs, rhs in generator:
+            yield (
+                test_name,
+                test_description,
+                cuspatial.GeoSeries(
+                    [
+                        lhs,
+                        rhs if types[0] == types[1] else lhs,
+                        lhs,
+                    ]
+                ),
```

Comment:
Is because it constructs a geoseries of 3 columns of mixed geometry types that computes mixed geometry columns binary predicates? Would be nice to explain it in docstring.

Reply:
I'll write a docstring for this and let you know when it is finished.

Reply:
Please have a look.